subsys/bluetooth/host/audio/capabilities.c

```diff
@@ -34,107 +34,227 @@ static sys_slist_t srcs;
  */
 static int unicast_server_config_cb(struct bt_conn *conn,
 				    const struct bt_audio_ep *ep,
-				    const struct bt_audio_capability *cap,
+				    uint8_t type,
 				    const struct bt_codec *codec,
-				    struct bt_audio_stream **stream)
+				    struct bt_audio_stream **stream,
+				    struct bt_codec_qos_pref *const pref)
 {
-	if (cap->ops == NULL || cap->ops->config == NULL) {
-		return -EACCES;
+	struct bt_audio_capability *cap;
+	sys_slist_t *lst;
+
+	if (type == BT_AUDIO_SINK) {
+		lst = &snks;
+	} else if (type == BT_AUDIO_SOURCE) {
+		lst = &srcs;
+	} else {
+		BT_ERR("Invalid endpoint type: %u", type);
+		return -EINVAL;
 	}
 
-	*stream = cap->ops->config(conn,
-				   (struct bt_audio_ep *)ep,
-				   (struct bt_audio_capability *)cap,
-				   (struct bt_codec *)codec);
+	SYS_SLIST_FOR_EACH_CONTAINER(lst, cap, node) {
+		/* Skip if capabilities don't match */
+		if (codec->id != cap->codec->id) {
+			continue;
+		}
+
+		if (cap->ops == NULL || cap->ops->config == NULL) {
+			return -EACCES;
+		}
 
-	if (*stream == NULL) {
-		return -ENOMEM;
+		*stream = cap->ops->config(conn, (struct bt_audio_ep *)ep,
+					   cap, (struct bt_codec *)codec);
+
+		if (*stream == NULL) {
+			return -ENOMEM;
+		}
```

Comment:
Do we think anyone could add multiple capabilities with same codec but different in other ways?
For example low quality CONTEXT_RINGTONE or CONTEXT_MAN_MACHINE with low/restircted SDU settings to allow coexistence with another ongoing CIS/BIS? And then another entry for high quality like media.
If that is the case I think this algorithm needs to compare more entries of the PAC before selecting which one to call-back on.

Reply:
That's a good question. The use case is seemingly there, but will need to consult the spec if we can even expose multiple PAC records with the same codec ID but with different config/meta data.